core/src/OpenACC/Kokkos_OpenACC_Instance.cpp

```diff
@@ -44,45 +44,43 @@
 
 #define KOKKOS_IMPL_PUBLIC_INCLUDE
 
-#include <OpenACC/Kokkos_OpenACC_Instance.hpp>
 #include <OpenACC/Kokkos_OpenACC.hpp>
-#include <OpenACC/Kokkos_OpenACC_Traits.hpp>
+#include <OpenACC/Kokkos_OpenACC_Instance.hpp>
 #include <impl/Kokkos_Profiling.hpp>
 #include <impl/Kokkos_DeviceManagement.hpp>
 
 #include <openacc.h>
 
 #include <iostream>
 
-namespace Kokkos {
-bool show_warnings() noexcept;
-}
+// Arbitrary value to denote that we don't know yet what device to use.
+int Kokkos::Experimental::Impl::OpenACCInternal::m_acc_device_num = -1;
 
-Kokkos::Experimental::Impl::OpenACCInternal*
+Kokkos::Experimental::Impl::OpenACCInternal&
 Kokkos::Experimental::Impl::OpenACCInternal::singleton() {
   static OpenACCInternal self;
-  return &self;
+  return self;
+}
+
+bool Kokkos::Experimental::Impl::OpenACCInternal::verify_is_initialized(
+    const char* const label) const {
+  if (!m_is_initialized) {
+    Kokkos::abort((std::string("Kokkos::Experimental::OpenACC::") + label +
+                   " : ERROR device not initialized\n")
+                      .c_str());
+  }
+  return m_is_initialized;
```

Comment:
I am not sure about the function name if we abort if uninitialized. Maybe we should call it 'abort_if_not_initialized' and return void.

Reply:
I plan on revisiting that later.  At the moment just following the other backend current implementation.  Let us merge as is for now.

Reply:
fine with me